include/deal.II/base/quadrature.h

```diff
@@ -221,6 +221,19 @@ class Quadrature : public Subscriptor
   template <class Archive>
   void serialize (Archive &ar, const unsigned int version);
 
+  /**
+   * This function returns true if the quadrature object is a tensor product
+   * of one-dimensional formulas.
+   */
+  bool is_tensor_product() const;
+
+  /**
+   * In case the quadrature formula is a tensor product, this function
+   * returns the one-dimensional basis objects.
+   * Otherwise, calling this function is not allowed.
+   */
+  const std::vector<Quadrature<1>> &get_tensor_basis() const;
```

Comment:
Ah, I see. I can think of ways around this, but it's awkward for sure. (The problem is not with the function, but with the member variable where you can't use `Quadrature<1>` in the `Quadrature<1>` class because that would be recursive. But you can use `Quadrature<1>` in `Quadrature<dim>`, and that would allow for some template tricks. That said, it's unnecessary complexity.)

But, thinking about it some, you could use a workaround in that the class has a member variable `std::unique_ptr<std::array<Quadrature<1>,dim>` which happens to be `null_ptr` in 1d or if the quadrature object is not a tensor product, and is set if the quadrature formula is a tensor product. Then the function here can just return a reference to the `std::array`. Furthermore, you don't have to explicitly store whether the object is a tensor product -- the condition is equal to `dim==1 || tensor_base != null_ptr`.